bot/exts/fun/off_topic_names.py

```diff
@@ -117,25 +152,133 @@ async def delete_command(self, ctx: Context, *, name: OffTopicName) -> None:
+        """
+        Re-rolls off topic name for an off-topic channel and blacklists the name.
+
+        ot_channel_index: [0, 1, 2, ...]
+        """
+        if ot_channel_index:
+            try:
+                channel = self.bot.get_channel(CHANNELS[ot_channel_index])
+            except IndexError:
+                await ctx.send(f":x: Off-topic channel not found with index {ot_channel_index}.")
+                return
+        elif ctx.channel.id in CHANNELS:
+            channel = ctx.channel
+
+        else:
+            await ctx.send("Please specify channel for which the off topic name should be re-rolled.")
+            return
+
+        old_channel_name = channel.name
+        old_ot_name = old_channel_name[NAME_START_INDEX:]  # ot1-name-of-ot -> name-of-ot
+
+        await self.de_activate_ot_name(ctx, old_ot_name)
+
+        response = await self.bot.api_client.get(
+            'bot/off-topic-channel-names', params={'random_items': 1}
+        )
+        try:
+            new_channel_name = response[0]
+        except IndexError:
+            await ctx.send("Out of active off topic names. Add new names to reroll.")
+            return
+
+        async def rename_channel() -> None:
+            """Rename off topic channel and log events."""
+            await channel.edit(
+                name=OTN_FORMATTER.format(number=old_channel_name[OT_NUMBER_INDEX], name=new_channel_name)
+            )
+            log.info(
+                f"{ctx.author} Off-topic channel re-named from `{old_ot_name}` "
+                f"to `{new_channel_name}`."
+            )
+
+            await ctx.message.reply(
+                f":ok_hand: Off-topic channel re-named from `{old_ot_name}` "
+                f"to `{new_channel_name}`. "
+            )
+
+        try:
+            await asyncio.wait_for(
+                rename_channel(),
+                3
+            )
+        except asyncio.TimeoutError:
+            # Channel rename endpoint rate limited. Cancel task and blacklist/de-activate name.
+            btn_yes = Button(label="Yes", style=ButtonStyle.success)
+            btn_no = Button(label="No", style=ButtonStyle.danger)
+
+            embed = Embed(
+                title=random.choice(NEGATIVE_REPLIES),
+                description=(
+
+                    "Re-naming the channel is being rate-limited. "
+                    "Would you like to schedule a channel re-name process within the current bot session ?"
+                ),
+                colour=Colour.blurple()
+            )
+
+            async def btn_call_back(schedule: bool, interaction: Interaction) -> None:
+
+                message = interaction.message
+
+                embed.description = (
+                    "Scheduled a channel re-name process within the current bot session."
+                    if schedule
+                    else
+                    "Channel not re-named due to rate limit. Please try again later."
+                )
```

Comment:
Do we want to be correct with terms here? It's an untracked asyncio Task - not a process.